public/app.js

```diff
@@ -19,15 +19,31 @@ class Controller {
         document.querySelector(DOMstrings.graphContainer).innerHTML = "";
     }
 
+    static displayCodingProgress() {
+        // Add the coding progress section to the UI
+        UIController.addCodingProgressSection();
+        // Get data for coding progress table
+        let unsubscribeFunc = DataController.watchCodingProgress(UIController.updateProgressUI);
+        DataController.registerSnapshotListener(unsubscribeFunc);
+    }
+
+    static displayProject(project) {
+        // Add the graphs container to the UI
+        UIController.addGraphs(project);
+        // Update and show the Graphs
+        let unsubscribeFunc = DataController.watchProjectTrafficData(
+            project,
+            GraphController.updateGraphs
+        );
+        DataController.registerSnapshotListener(unsubscribeFunc);
+    }
+
     static navigateToCodingProgress(e) {
         if (e.target && e.target.nodeName == "A") {
             Controller.resetUI();
             DataController.detachSnapshotListener();
-            // Add the coding progress section to the UI
-            UIController.addCodingProgressSection();
-            // Get data for coding progress table
-            let unsubscribeFunc = DataController.watchCodingProgress(UIController.updateProgressUI);
-            DataController.registerSnapshotListener(unsubscribeFunc);
+            location.hash = "coding_progress";
```

Comment:
Can this be window.location.hash?

(You read from window.location.hash, and I think that's a clearer name)